src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -39,7 +43,7 @@ public class FolderFileStorage : IFileStorage {
 
             try {
                 return Task.FromResult<Stream>(File.OpenRead(Path.Combine(Folder, path)));
-            } catch (FileNotFoundException) {
+            } catch (IOException ex) when(ex is FileNotFoundException || ex is DirectoryNotFoundException) {
                 return Task.FromResult<Stream>(null);
```

Comment:
We should log a trace message here just so we can see more information if needed